src/viking.js

```diff
@@ -1,14 +1,86 @@
 // Soldier
-class Soldier {}
+class Soldier {
+  constructor (health, strength) {
+    this.health = health;
+    this.strength = strength;
+  }
+  attack(){
+    return this.strength;
+  };
+  receiveDamage(damage) {
+    this.health -= damage;
+  } ;
+}
+
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+  constructor (name, health, strength) {
+    super (health, strength);
+    this.name = name;
+  }
+  receiveDamage(damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`
+    } else {
+      return `${this.name} has died in act of combat`
+    }
+  }
+  battleCry () {
+    return "Odin Owns You All!"
+  }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+  receiveDamage (damage) {
+    this.health -= damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`
+    } else {
+      return `A Saxon has died in combat`
+    } 
+  }
+}
 
 // War
-class War {}
+class War {
+  constructor () {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  
+  addViking (viking) {
+    this.vikingArmy.push(viking);
+  }
+
+  addSaxon (saxon) {
+    this.saxonArmy.push(saxon);
+  }
+
+  vikingAttack () {
+    let saxon = saxon.this.saxonArmy[Math.floor(Math.random()*this.saxonArmy.length)];
+    saxon = new Saxon();
```

Comment:
You don't need to add the Saxons as the tests already does that. And the same goes for vikings :)